sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -470,6 +470,52 @@ object BooleanSimplification extends Rule[LogicalPlan] with PredicateHelper {
   }
 }
 
+/**
+ * Optimize Rand expressions in binary comparison:
+ * 1. Converts the binary comparison to true when the comparison value must be true.
+ * 2. Converts the binary comparison to false when the comparison value must be false.
+ */
+object OptimizeRand extends Rule[LogicalPlan] with PredicateHelper {
+  val oneLiteral = Literal(1d)
+
+  def apply(plan: LogicalPlan): LogicalPlan = plan.transformWithPruning(
+    _.containsPattern(EXPRESSION_WITH_RANDOM_SEED), ruleId) {
+    case q: LogicalPlan => q.transformExpressionsUpWithPruning(
+      _.containsAnyPattern(EXPRESSION_WITH_RANDOM_SEED), ruleId) {
+      case GreaterThan(left, Rand(_, _))
```

Comment:
we can match DoubleLiteral directly. Other optimizer rules will optimize foldable expressions to literals.

Reply:
Thank you for the reminder.